index.bs

```diff
@@ -2360,53 +2372,60 @@ The entry key MUST be the extension identifier, and the value MUST be the [=clie
 
 Note: Extensions should aim to define authenticator arguments that are as small as possible. Some authenticators communicate
     over low-bandwidth links such as Bluetooth Low-Energy or NFC.
 
 
-## Extending <dfn>client processing</dfn> ## {#extension-client-processing}
+## <dfn>Client extension processing</dfn> ## {#sctn-client-extension-processing}
 
 Extensions may define additional processing requirements on the client platform during the creation of credentials or the
 generation of an assertion. In order for the [=[RP]=] to verify the processing took place, or if the processing has a result
-value that the [RP] needs to be aware of, the extension should specify a key-value pair to be included in the [=client data=].
+value that the [RP] needs to be aware of, the extension specifies a extension key-value pair to be included in the [=client data=] extensions field.
 
-The [=client data=] value may be any value that can be encoded using JSON. If any extension processed by a client defines such a
+If any extension processed by a client defines such a
 value, the client SHOULD include a dictionary in its [=client data=] with the key {{CollectedClientData/extensions}}. For each such
-extension, the client SHOULD add an entry to this dictionary with the extension identifier as the key, and the extension's
-[=client data=] value.
+extension, the client SHOULD add an entry to this dictionary with the [=extension identifier=] as the key, and the extension's
+[=client extension input=] value.
+The [=client extension input=] value may be any value that can be encoded using JSON.
 
-Extensions that require authenticator processing MUST define the process by which the client argument can be used to determine
-the authenticator argument.
+Likewise, the extension output is represented in the [=client data=] as a CBOR map
+with [=extension identifier=] values as keys, and the <dfn>client extension output</dfn> value of each extension as the value.
 
+Extensions that require authenticator processing MUST define
+the process by which the [=client extension input=] can be used to determine the [=authenticator extension input=] and
+the process by which the [=authenticator extension output=] can be used to determine the [=client extension output=].
 
-## Extending authenticator processing ## {#extension-authenticator-processing}
+## <dfn>Authenticator extension processing</dfn> ## {#sctn-authenticator-extension-processing}
 
-Extensions that define additional authenticator processing may similarly define an [=authenticator data=] value. The value may
+Extensions that define additional authenticator processing define an [=authenticator extension input=] value. The value may
 be any data that can be encoded in CBOR. An authenticator that processes an [=authentication extension=] that defines such a
-value must include it in the [=authenticator data=].
+value must include it in the [=authenticator data=] extensions section.
```

Comment:
it seems that this sentence..
> An authenticator that processes an [=authentication extension=] that defines such a value must include it in the [=authenticator data=] extensions section.

..and this sentence..
> As specified in [[#sec-authenticator-data]], the [=authenticator extension input=] value of each processed extension is included in the
-extended data part of the [=authenticator data=]. This part is a CBOR map, with extension identifiers as keys, and the
+extensions data part of the [=authenticator data=].

..are redundant?